include/exadg/time_integration/time_integration_constants_base.h

```diff
@@ -0,0 +1,113 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_TIME_INTEGRATION_TIME_INTEGRATION_CONSTANTS_BASE_H_
+#define INCLUDE_EXADG_TIME_INTEGRATION_TIME_INTEGRATION_CONSTANTS_BASE_H_
+
+
+namespace ExaDG
+{
+class TimeIntegratorConstantsBase
+{
+public:
+  TimeIntegratorConstantsBase(unsigned int const order, bool const start_with_low_order)
+    : order(order), start_with_low_order(start_with_low_order)
+  {
+  }
+
+  virtual ~TimeIntegratorConstantsBase()
+  {
+  }
+
+  /*
+   *  This function updates the time integrator constants in case of constant time step sizes.
+   */
+  void
+  update(unsigned int const current_order)
+  {
+    // when starting the time integrator with a low order method, ensure that
+    // the time integrator constants are set properly
+    if(current_order <= order && start_with_low_order == true)
+    {
+      set_constant_time_step(current_order);
+    }
+    else
+    {
+      set_constant_time_step(order);
+    }
+  }
+
+  /*
+   *  This function updates the time integrator constants in case of adaptive time step sizes.
+   */
+  void
+  update(unsigned int const current_order, std::vector<double> const & time_steps)
+  {
+    // when starting the time integrator with a low order method, ensure that
+    // the time integrator constants are set properly
+    if(current_order <= order && start_with_low_order == true)
+    {
+      set_adaptive_time_step(current_order, time_steps);
+    }
+    else // adjust time integrator constants since this is adaptive time stepping
+    {
+      set_adaptive_time_step(order, time_steps);
+    }
+  }
+
+  unsigned int
+  get_order() const
+  {
+    return order;
+  }
+
+  /*
+   *  This function prints the time integrator constants
+   */
+  virtual void
+  print(dealii::ConditionalOStream & pcout) const = 0;
+
+protected:
+  // order of time integrator
+  unsigned int const order;
```

Comment:
This can be private. We already have a getter.

Reply:
Yes. The problem is that the previous implementation did not make use of this getter. If we change it now, it will imply more changes also in #261. Principally, I am open to changing it here or separately.

Reply:
If we merge this first...
If you persist on this, I have to change https://github.com/exadg/exadg/pull/261 again anyway.

Reply:
It is not about "persisting". You do not see a value in the present PR (and doing it now before adding more code duplications) that removes more lines of code than it adds new lines of code?